### PR TITLE
feat(header): Use strings for authentication state

### PR DESCRIPTION
**Pull request by michaeltaranto**
BREAKING CHANGE: Moved away from true/false/null for tri-state management of `authenticated` on the `<Header />`. Now uses explicit strings of `authenticated`/`unauthenticated`/`pending` being passed into `authenticationStatus`.

### CODE REVIEW

File: react/Header/Header.js [hunk 1]
```diff
@@ -16,17 +16,22 @@ const employerLinkHref = locale => locale === 'NZ' ?
   'https://talent.seek.co.nz/' :
   'https://talent.seek.com.au/';
 
+const AUTHENTICATED = 'authenticated';
```
Comment (pokle) [inline react/Header/Header.js]: Can we have these constants exported?

Comment (markdalgleish) [inline react/Header/Header.js]: I'm not sure it's worth the effort, given we're using `propTypes`, and the API would be a bit awkward. The best I can come up with as a scalable pattern would look something like this, which I'm not a fan of:

```js
import { Header } from 'seek-style-guide/react';
const { AUTHENTICATED, UNAUTHENTICATED, PENDING } = Header.authenticatonStatus;
```

Comment (michaeltaranto) [inline react/Header/Header.js]: I looked into exporting these constants but it required a fair bit of boilerplate for the benefit returned (in terms of wiring up @markdalgleish's code above to the PropTypes). I would prefer this to go in consistently for now, but we discuss how to manage this pattern going forward

Comment (pokle) [inline react/Header/Header.js]: Agreed, not worth the effort.

**Pull request merged**